File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/following/PrecisionModeThresholds.java
package org.firstinspires.ftc.teamcode.following;

public class PrecisionModeThresholds {

    public final double entryPositionDistance, exitPositionDistance;
    public final double entryVelocity, exitVelocity;
    public final double entryHeadingError, exitHeadingError;
    public final double entryAngularVelocity, exitAngularVelocity;

    public PrecisionModeThresholds(
            double entryPositionDistance, double exitPositionDistance,
            double entryVelocity, double exitVelocity,
            double entryHeadingError, double exitHeadingError,
            double entryAngularVelocity, double exitAngularVelocity
    ) {

        this.entryPositionDistance = entryPositionDistance;
        this.exitPositionDistance = exitPositionDistance;

        this.entryVelocity = entryVelocity;
        this.exitVelocity = exitVelocity;

        this.entryHeadingError = entryHeadingError;
        this.exitHeadingError = exitHeadingError;

        this.entryAngularVelocity = entryAngularVelocity;
        this.exitAngularVelocity = exitAngularVelocity;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/following/PrecisionModeThresholds.kt
package org.firstinspires.ftc.teamcode.following

data class PrecisionModeThresholds(
    val entryPositionDistance: Double,
    val exitPositionDistance: Double,
    val entryVelocity: Double,
    val exitVelocity: Double,
    val entryHeadingError: Double,
    val exitHeadingError: Double,
    val entryAngularVelocity: Double,
    val exitAngularVelocity: Double
)